### PR TITLE
fix(Minimap): no free available atlas positions

### PR DESCRIPTION
**Pull request by MukjepScarlet**
Hope it works...

### CODE REVIEW

File: src/main/kotlin/net/ccbluex/liquidbounce/integration/theme/component/types/minimap/MinimapTextureAtlasManager.kt [hunk 1]
```diff
@@ -40,15 +42,18 @@ private const val ATLAS_SIZE: Int = 64
...
 
 class MinimapTextureAtlasManager {
     private val texture = NativeImageBackedTexture(ATLAS_SIZE * 16, ATLAS_SIZE * 16, false)
-    private val availableAtlasPositions = ArrayList<AtlasPosition>(ATLAS_SIZE * ATLAS_SIZE - 1)
+    private val availableAtlasPositions = ArrayList<AtlasPosition>(MAX_ATLAS_POSITIONS)
     private val dirtyAtlasPositions = hashSetOf<AtlasPosition>()
     private val chunkPosAtlasPosMap = hashMapOf<ChunkPos, AtlasPosition>()
 
     private val lock = ReentrantReadWriteLock()
+    private val semaphore = Semaphore(MAX_ATLAS_POSITIONS)
```
Comment (superblaubeere27) [inline src/main/kotlin/net/ccbluex/liquidbounce/integration/theme/component/types/minimap/MinimapTextureAtlasManager.kt]: Why is this needed when we have `availableAtlasPositions`?

**Review by superblaubeere27 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/net/ccbluex/liquidbounce/integration/theme/component/types/minimap/MinimapTextureAtlasManager.kt (hunk 1 above)
Comment (MukjepScarlet) [reply src/main/kotlin/net/ccbluex/liquidbounce/integration/theme/component/types/minimap/MinimapTextureAtlasManager.kt]: The `removeLastOrNull` could be called when other threads(coroutines) are putting items back, so an exception will occur, we should wait until at least one item is available

Comment (superblaubeere27) [reply src/main/kotlin/net/ccbluex/liquidbounce/integration/theme/component/types/minimap/MinimapTextureAtlasManager.kt]: But that would cause a thread to be stalled as soon as there is no more space in the Minimap texture atlas which is not a good method of handling failure :/

I will probably just add a proper fix haha

**Comment by DataM0del** [comment]
I think this would fix my PR for merging upstream into the 1.21.3 branch (it's #4722)

File: src/main/kotlin/net/ccbluex/liquidbounce/integration/theme/component/types/minimap/MinimapTextureAtlasManager.kt (hunk 1 above)
Comment (DataM0del) [reply src/main/kotlin/net/ccbluex/liquidbounce/integration/theme/component/types/minimap/MinimapTextureAtlasManager.kt:56]: > But that would cause a thread to be stalled as soon as there is no more space in the Minimap texture atlas which is not a good method of handling failure :/
> 
> I will probably just add a proper fix haha

I'll wait for it, I guess...

Comment (MukjepScarlet) [reply src/main/kotlin/net/ccbluex/liquidbounce/integration/theme/component/types/minimap/MinimapTextureAtlasManager.kt]: I estimate that the scanning thread will be blocked for a few seconds at most, and the main thread will not be affected. Another choice is to change the functions (of Subscribers) to suspend functions...

**Review by 1zun4 — APPROVED** [review]
